File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input = sys.stdin.readline

N = int(input())
A = list(map(int, input().split()))
A.sort() #이진 탐색을 위한 정렬
M = int(input())
B = list(map(int, input().split()))

# 1. A 배열 정렬하기
# 2. B에 든 M개의 배열 하나씩 보면 서, A에서 이진 탐색하기

def search(st,en,target):
	if st==en:
		# 가장 마지막 나눈 범위임. 이때 값이 있으면 T 없으면 F가 됨.
		if A[st] == target:
			print("1")
		else:
			print("0")
		return
		
	mid = (st+en)//2
	if A[mid]<target:
		search(mid+1,en,target)
	else:
		search(st,mid,target)
		
		
for num in B:
	search(0, N-1, num)

File: 백준/Silver/1920. 수 찾기/수 찾기.java
/*

*/
import java.util.*;
import java.io.*;

class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb= new StringBuilder();
        
        int N = Integer.parseInt(br.readLine());
        Set<Integer> A = new HashSet<>();
        st = new StringTokenizer(br.readLine());
        for(int i=0; i<N; i++){
           A.add(Integer.parseInt(st.nextToken()));
        }
        
        int M = Integer.parseInt(br.readLine());
        for(String s: br.readLine().split(" ")){
            int num = Integer.parseInt(s);
            
            if(A.contains(num)){
                sb.append(1).append("\n");
            }else{
                sb.append(0).append("\n");
            }
        }
        
        System.out.println(sb.toString().trim());
    }
}